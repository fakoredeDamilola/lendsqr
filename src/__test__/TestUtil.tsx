
import { render } from "@testing-library/react";
import { Provider } from "react-redux";

export function testRender(jsx:any, { store, ...otherOpts }:{store:any}) {
  return render(
  <Provider store={store}>
    {jsx}
    </Provider>
    , otherOpts);
}

