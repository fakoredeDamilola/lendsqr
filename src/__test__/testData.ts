
const testObj1 = {
    name:"Fakorede",
    company:"data",
    text:"you are"

}
const testObj2 = {
    name:"",
    company:"data",
    text:"you are"

}
const testObj3 = {
    name:"",
    company:"",
    text:""

}

const user1={
    "createdAt": "2093-05-10T11:18:24.966Z",
    "orgName": "dolorem-totam-et",
    "userName": "Vickie_Ruecker",
    "email": "Bertrand99@yahoo.com",
    "phoneNumber": "(742) 411-7858 x1596",
    "lastActiveDate": "2099-05-17T17:03:00.657Z",
    "profile": {
        "firstName": "Shayne",
        "lastName": "Hickle",
        "phoneNumber": "(772) 391-9108 x282",
        "avatar": "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/669.jpg",
        "gender": "Male",
        "bvn": "143400664",
        "address": "Alexanne Summit",
        "currency": "NGN"
    },
    "guarantor": {
        "firstName": "Isaiah",
        "lastName": "Macejkovic",
        "phoneNumber": "510.315.0223 x739",
        "gender": "Male",
        "address": "Maxwell Rest"
    },
    "accountBalance": "331.34",
    "accountNumber": "UGFUHNC1",
    "socials": {
        "facebook": "@lendsqr",
        "instagram": "@lendsqr",
        "twitter": "@lendsqr"
    },
    "education": {
        "level": "Bsc",
        "employmentStatus": "Employed",
        "sector": "FinTech",
        "duration": "2 Years",
        "officeEmail": "Vernice.Fisher56@gmail.com",
        "monthlyIncome": [
            "701.36",
            "881.56"
        ],
        "loanRepayment": "530.77"
    },
    "id": "44"
}

export {testObj1,testObj2,testObj3,user1}