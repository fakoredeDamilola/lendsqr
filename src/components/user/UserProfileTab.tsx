import "../../styles/user.scss"

interface IProfileItem {
    name:string;
    value?:string
  }
  
  interface IProfileTab {
    heading:string;
    profileInfoItem:IProfileItem[]
  }

const UserProfileTab = ({profileItem}:{profileItem:IProfileTab}) => {
  return (
    <div className='user_profile_tab'>
    <h3>{profileItem.heading}</h3>
    <div className="user_profile_tab_info">
        {profileItem.profileInfoItem.map((infoItem:IProfileItem,index:number)=>{
            return (
                <div key={index}>
                <h6>{infoItem.name.toUpperCase()}</h6>
                    <p>{infoItem.value}</p>
                </div>
            )
        })

        }
    </div>
    </div>
  )
}

export default UserProfileTab