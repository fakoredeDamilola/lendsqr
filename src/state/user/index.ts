import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { IUser } from "../../utils/interface";



export enum IUserTab {
    GENERAL_DETAILS="General Details",
    DOCUMENTS="Documents",
    BANK_DETAILS="Bank Details",
    LOANS="Loans" ,
    SAVINGS="Savings",
    APP_SYSTEM="App and System"
}

export const searchFilterDropdown ={
    "organization":[
      {name:"organization",value:"org1"},
      {name:"organization",value:"org2"},
      {name:"organization",value:"org3"},
    ],
    "status":[
      {name:"status",value:"stat1"},
      {name:"status",value:"stat2"},
      {name:"status",value:"stat3"},
    ]
   }

interface UserState {
    users : IUser[];
    filterSearch : IUser[];
    user: IUser | null;
    currentPage: number;
    userPerPage:number;
    userListOnPage:IUser[]
    currentSideTab:string;
}

export const UserTabs = ["General Details","Documents","Bank Details","Loans","Savings","App and System"]

const initialState : UserState = {
    users: [ ],
    filterSearch:[],
    user:null,
    userPerPage:10,
    currentPage:1,
    userListOnPage:[],
    currentSideTab:"users",
   
       
}
const userSlice = createSlice({
    name : "user",
    initialState,
   reducers: {
    updateUsers: (state , action : PayloadAction<{users : IUser[]}>) => {
        state.users = action.payload.users
        state.filterSearch = action.payload.users
        
        const startIndex = state.currentPage ===1 ? 0 : state.currentPage * state.userPerPage
        const endIndex = state.currentPage * state.userPerPage +10
        const miniUsers = action.payload.users.slice(startIndex,endIndex)
        state.userListOnPage = miniUsers
    },
    updateUser: (state , action : PayloadAction<{user : IUser | null}>) => {
        
        state.user = action.payload.user
    },
    changeUserPage: (state , action : PayloadAction<{page : number,users:IUser[]}>) =>{
        const startIndex = action.payload.page ===1 ? 0 : (action.payload.page-1) * state.userPerPage
        const endIndex = (action.payload.page * state.userPerPage) + 1
        const miniUsers =action.payload.users.slice(startIndex,endIndex)
        state.userListOnPage = miniUsers
        state.currentPage=action.payload.page
    },
    switchSideBarTab: (state , action : PayloadAction<{sidetab:string}>) => {
        state.currentSideTab = action.payload.sidetab
    },
    changeuserPerPage: (state , action : PayloadAction<{userPerPage:number,users:IUser[]}>) => {
        state.userPerPage= action.payload.userPerPage
        const startIndex = state.currentPage ===1 ? 0 : (state.currentPage-1) * state.userPerPage
        const endIndex = (state.currentPage * state.userPerPage) + 1
        const miniUsers =action.payload.users.slice(startIndex,endIndex)
        state.userListOnPage = miniUsers
        state.currentPage=state.currentPage
    },
    changeFilterSearch:(state , action : PayloadAction<{search:string,users:IUser[],main_users:IUser[]}>) =>{
        const {search,users,main_users} = action.payload
        if(search){
            state.filterSearch = users
            
        }else{
            state.filterSearch = main_users
        }
        const startIndex = state.currentPage ===1 ? 0 : (state.currentPage-1) * state.userPerPage
            const endIndex = (state.currentPage * state.userPerPage) + 1
            const miniUsers =users.slice(startIndex,endIndex)
            state.userListOnPage = miniUsers
            state.currentPage=state.currentPage
    }
   }
});

export const  { updateUsers,changeUserPage,updateUser,switchSideBarTab,changeuserPerPage,changeFilterSearch }  = userSlice.actions
export default userSlice.reducer;
