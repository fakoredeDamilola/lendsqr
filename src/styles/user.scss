@use 'utilities'; 

.user_page{
    
}

.user_details_section{
    display:flex;
    justify-content: space-between;
}
.user_details_button{
    display:flex;
    gap:20px;
    button{
        width:170px;
        height:40px;
        background:transparent;
        border-radius:calc(utilities.$border_radius*2) ;

        cursor:pointer;
    }  
    .user_activate_button{
        border: 1px solid utilities.$secondary_color;
        color: utilities.$secondary_color;
        transition: 0.3s all;
        &:hover{
            background:utilities.$secondary_color;
            color:white;
        }
    }
    .user_blacklist_button{
        border:1px solid #E4033B;
        color:#E4033B;
        transition: 0.3s all;
        &:hover{
            background:#E4033B;
            color:white;
        }
    }
}
.section{
    padding:20px;
    box-shadow: 3px 5px 20px rgba(0, 0, 0, 0.04);
    box-sizing:border-box;
    width:100%;
    border-radius: utilities.$border_radius;
    background:#FFFFFF;
    position:relative;
    
}
.user_general_section{
    padding-bottom: 0;
    .user_general_info_section{
    align-items: center;
    }
    .user_tier{
        display:flex !important;
        // justify-content: space-between;
        // background:red;
    }
}

.user_section_one_text{
    display: flex;
    flex-direction: column;
    gap:30px;
    padding: 0px;
    margin: 15px 0;
    box-sizing: border-box;
}
.user_section_one_text:nth-child(3){
    border:none;
}
.user_section_one_tabs{
    display:flex;
    padding-top:20px;
    width:100%;
    overflow-x: scroll;
    justify-content: space-between;
    border-bottom: 2px solid transparent;
    .selected{
        border-bottom:2px solid utilities.$secondary_color;
        color:utilities.$secondary_color;
    }
    .user_section_one_tab{
        flex-shrink: 0;
        padding:0 20px;
    padding-bottom: 3px;
    cursor:pointer;
    
    &:hover {
        border-bottom:2px solid utilities.$secondary_color;
        color:utilities.$secondary_color;

    }
    }
}
.user_not_found{
    padding:40px 0;
    display:flex;
    justify-content: center;
    align-items: center;
    flex-direction:column;
    div {
        margin:20px 0;
    }
  }
.user_profile_tab{

    // opacity: 0.1;
    padding:20px 0;
    border-top: 1px solid #213F7D;
  .user_profile_tab_info{ 
    display:grid;
    grid-template-columns:repeat(auto-fit, minmax(220px,1fr)); 
    grid-row-gap:25px;
   margin:25px 0;
    p {
        font-size:16px;
        font-weight: 500;
        margin-top: 15px;
    }
}
}
.not-found-tab{
    padding:50px 0;
    text-align: center;
}
.desktop_button{
    display: none;
}
.mobile_button{
    margin:20px 0;
}

@media (min-width: utilities.$breakpoint-tablet) {
    .user_general_info_section{
          display:flex;
    }
    .user_section_one_text:nth-child(3){
    border:1px solid #545F7D;
    border-top:none;
    border-bottom:none;
}
.user_section_one_text{
    padding:0 40px;
    margin: 15px 0;
}
.desktop_button{
    display: block;
}
.mobile_button{
    display: none;
}
.user_profile_tab_info{
    grid-template-columns:repeat(4,1fr);
     
}
}
 