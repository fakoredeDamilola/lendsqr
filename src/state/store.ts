import { configureStore, getDefaultMiddleware } from "@reduxjs/toolkit";
import { save, load } from "redux-localstorage-simple";
import user from "./user";

const PERSISTED_KEYS: string[] = [];

const store = configureStore({
  reducer: {
    user,
  },
  middleware: [
    ...getDefaultMiddleware({
      serializableCheck: false,
      thunk: true,
    }),
    // save({ states: PERSISTED_KEYS }),
  ],
//   preloadedState: load({ states: PERSISTED_KEYS }),
});

export default store;

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
