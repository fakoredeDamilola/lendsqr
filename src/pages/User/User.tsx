import React, { useState,useEffect } from 'react'
import BackLink from '../../components/BackLink'
import "../../styles/user.scss"
import CustomProfilePicture from '../../components/custom/CustomProfilePicture'
import { numberWithCommas } from '../../utils/utilFunction'
import { IUserTab, UserTabs, updateUser } from '../../state/user'
import UserProfileTab from '../../components/user/UserProfileTab'
import { Link, useParams } from 'react-router-dom'
import { useDispatch, useSelector } from 'react-redux'
import { RootState } from '../../state/store'
import { IUser } from '../../utils/interface'
import NotFound from '../../components/NotFound'

const User = () => {

  const {users,user} = useSelector((state:RootState)=>state.user)
  const { id } = useParams();
  const dispatch = useDispatch()

const [loading,setLoading] = useState(true)
const [currentTab,setCurrentTab] = useState("Documents")

const fetchUserData = async () =>{
  const data = await fetch(`https://6270020422c706a0ae70b72c.mockapi.io/lendsqr/api/v1/users/${id}`)
  const response:IUser = await data.json()
  console.log({response})
  return response
  
 }


useEffect(()=>{
const getUser = async ()=>{
const user =users.length> 0 ? 
            users.find((user:IUser) => user.id === id) :
           await fetchUserData()
  dispatch(updateUser({user:user ?? null})) 
  setLoading(false)
  }
  getUser()
},[id])


console.log({user})
const blackListUser = () =>{

}
const userBasicInfo={
  fullName:`${user?.profile.firstName} ${user?.profile.lastName}`,
  orgName: user?.orgName,
  tier:3,
  accountBalance:user?.accountBalance,
  
  bankNumber:user?.accountNumber,
  bankName:"meekmmk"
}
interface IProfileItem {
  name:string;
  value?:string
}

interface IProfileTab {
  heading:string;
  profileInfoItem:IProfileItem[]
}
const userProfileInfo:IProfileTab[] = [
  {
    heading:"Personal Information",
    profileInfoItem: [
      {
        name:"full name",
        value:`${user?.profile.firstName} ${user?.profile.lastName}`
      },
      {
        name:"phone number",
        value: user?.profile?.phoneNumber
      },
      {
        name:"email address",
        value: user?.email
      },
      {
        name:"bvn",
        value: user?.profile.bvn
      },
      {
        name:"gender",
        value: user?.profile.gender
      },
      {
        name:"marital status",
        value: "Single"
      },
      {
        name:"children",
        value: "None"
      },
      {
        name:"type of residence",
        value: user?.profile?.address
      },

    ]
  },
  {
    heading:"Education and Employment",
    profileInfoItem: [
      {
        name:"level of education",
        value:user?.education?.level
      },
      {
        name:"employment status",
        value:user?.education?.employmentStatus
      },
      {
        name:"sector of employment",
        value:user?.education?.sector
      },
      {
        name:"duration of employment",
        value:user?.education?.duration
      },
      {
        name:"office mail",
        value:user?.education?.officeEmail
      },
      {
        name:"monthly income",
        value:user?.education?.monthlyIncome.map((income)=>`₦${income}`).join(" - ")
      },
      {
        name:"loan repayment",
        value:user?.education?.loanRepayment
      },
    ]
  },
  {
    heading:"Socials",
    profileInfoItem: [
      {
        name:"twitter",
        value:user?.socials?.twitter,
      },
      {
        name:"facebook",
        value:user?.socials?.facebook,
      },
      {
        name:"instagram",
        value:user?.socials?.instagram,
      },
    ]
  },
  {
    heading:"guarantor",
    profileInfoItem:[
      {
        name:"full name",
        value:`${user?.guarantor?.firstName} ${user?.guarantor?.lastName}`
      },
      {
        name:"phone number",
        value:`${user?.guarantor?.phoneNumber}`
      },
      {
        name:"email address",
        value:`${user?.guarantor.address}`
      },
      {
        name:"relationship",
        value:`${user?.guarantor.gender}`
      },
    ]
  }
]
    return (
        <div className="user_page">
        <h2>Users</h2>
        <main className="user_page_main">
        <section className="user_page_section">
         <BackLink links={[{href:"dashboard",text:"Back to Users"}]} />
        </section>
          
        {user ? 
        <div>
<section className='user_details_section'>
          <div className="user_details">
            <h2>User Details</h2>
          </div>
          <div className="user_details_button desktop_button">
            <button onClick={blackListUser} className='user_blacklist_button'>BLACKLIST USER</button>
            <button onClick={blackListUser} className='user_activate_button'>ACTIVATE USER</button>
          </div>
         </section>
         <section className="section user_general_section">
          <div className="user_general_info_section">
             <CustomProfilePicture image={user?.profile?.avatar} name="" style={{width:"100px",height:"100px"}} size="40px" color="blue" />
          <div className="user_section_one_text">
        <h3>{userBasicInfo.fullName}</h3>
        <p>{userBasicInfo.orgName}</p>
          </div>
          <div className="user_section_one_text user_tier">
        <p>User’s Tier</p>
        <p>{userBasicInfo.tier} stars</p>
          </div>
          <div className="user_section_one_text">
        <h3>N{numberWithCommas(userBasicInfo.accountBalance)}</h3>
        <p>{userBasicInfo.bankNumber}/{userBasicInfo.bankName}</p>
          </div>
          </div>
          <div className="user_details_button mobile_button">
            <button onClick={blackListUser} className='user_blacklist_button'>BLACKLIST USER</button>
            <button onClick={blackListUser} className='user_activate_button'>ACTIVATE USER</button>
          </div>
          <div className="user_section_one_tabs">
             {UserTabs.map((tab:string,index:number)=>{
            return (
              <div key={index} className={`user_section_one_tab ${currentTab === tab ?"selected":""}`} onClick={()=>setCurrentTab(tab)}>
                {tab}
              </div>
            )
          })

          }
          </div>
         
         </section>
         <section className=" section">
         {currentTab === IUserTab.GENERAL_DETAILS ?
          <div>
            {userProfileInfo.map((profileInfo:IProfileTab,index:number)=>{
              console.log({userProfileInfo})
              return (
                <UserProfileTab
                key={index} 
                profileItem={profileInfo}
                />
              )
            })

          }
          </div>:
          <div className='not-found-tab'>
           <h1>Coming soon</h1>
          </div>
          }
          
         </section>

        </div>
         : <div className="user_not_found">
         <NotFound title="No User Found" text="uh-oh....we understand what you feel" />
         <Link to="/dashboard">Go back home</Link>
       </div>}
        </main>
        
    
        </div>
      )
}

export default User