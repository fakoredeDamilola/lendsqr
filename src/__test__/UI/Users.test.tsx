import { render, screen, waitFor } from '@testing-library/react';
import User from '../../pages/User/User';
import { testRender } from "../TestUtil";
import wait from 'waait';
import store from "../../state/store"
import { BrowserRouter as Router } from "react-router-dom"
import { act } from 'react-dom/test-utils';
import { updateUser } from '../../state/user';
import { user1 } from '../testData';


describe('<User />',()=>{
  
  it("has the user text",async ()=>{
    const user=testRender(
    <Router>
      <User/>
    </Router>
    , { store, });
      const back = screen.getByText("Back to Users")
        expect(back).toBeInTheDocument()
        act(()=>{
          store.dispatch(updateUser({user:user1}));
          })
          
          await waitFor(() => wait(3));
        //   const fullname = screen.getByText("Shayne Hickle")
          const nouser= screen.getByText("No User Found")
          expect(nouser).toBeInTheDocument()
        //   expect(fullname).toBeInTheDocument()
      
  })
  it("Matches snapshot",()=>{
    const user=testRender(
      <Router>
        <User/>
      </Router>
      , { store, });
    expect(user).toMatchSnapshot()
})
})