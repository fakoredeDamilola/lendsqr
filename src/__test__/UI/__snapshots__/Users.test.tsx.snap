// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<User /> > Matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="user_page"
      >
        <h2>
          Users
        </h2>
        <main
          class="user_page_main"
        >
          <section
            class="user_page_section"
          >
            <div
              class="back_container"
            >
              <div
                class="back_icon"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M15 8a.5.5 0 0 0-.5-.5H2.707l3.147-3.146a.5.5 0 1 0-.708-.708l-4 4a.5.5 0 0 0 0 .708l4 4a.5.5 0 0 0 .708-.708L2.707 8.5H14.5A.5.5 0 0 0 15 8z"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
              <div
                class="back_link_texts"
              >
                <a
                  class="link_tag"
                  href="/dashboard"
                >
                  Back to Users
                </a>
              </div>
            </div>
          </section>
          <div
            class="user_not_found"
          >
            <div
              class="not_found"
            >
              <h1>
                No User Found
              </h1>
              <div
                class="not_found_text"
              >
                uh-oh....we understand what you feel
              </div>
            </div>
            <a
              href="/dashboard"
            >
              Go back home
            </a>
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="user_page"
    >
      <h2>
        Users
      </h2>
      <main
        class="user_page_main"
      >
        <section
          class="user_page_section"
        >
          <div
            class="back_container"
          >
            <div
              class="back_icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15 8a.5.5 0 0 0-.5-.5H2.707l3.147-3.146a.5.5 0 1 0-.708-.708l-4 4a.5.5 0 0 0 0 .708l4 4a.5.5 0 0 0 .708-.708L2.707 8.5H14.5A.5.5 0 0 0 15 8z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              class="back_link_texts"
            >
              <a
                class="link_tag"
                href="/dashboard"
              >
                Back to Users
              </a>
            </div>
          </div>
        </section>
        <div
          class="user_not_found"
        >
          <div
            class="not_found"
          >
            <h1>
              No User Found
            </h1>
            <div
              class="not_found_text"
            >
              uh-oh....we understand what you feel
            </div>
          </div>
          <a
            href="/dashboard"
          >
            Go back home
          </a>
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
